consensus/src/test/java/org/apache/iotdb/consensus/ratis/RatisConsensusTest.java

```diff
@@ -109,18 +118,74 @@ public DataSet read(IConsensusRequest IConsensusRequest) {
     }
 
     @Override
-    public void takeSnapshot(ByteBuffer metadata, File snapshotDir) {}
+    public void takeSnapshot(ByteBuffer metadata, File snapshotDir) {
+      String tempFilePath = snapshotDir + File.separator + ".tmp";
+      String filePath = snapshotDir + File.separator + "snapshot." + new String(metadata.array());
```

Comment:
When `IStateMachine` take the snapshot, it can directly use the metadata to name the snapshot file. It's guaranteed that more up-to-date snapshot will have lexicographically larger metadata.